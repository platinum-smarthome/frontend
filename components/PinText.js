import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'

class PinText extends Component {
  render() {
    return (
      <View style={styles.homeDiv}>
        <Text style={styles.homeTitle}>{this.props.text}</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  homeDiv: {
    alignItems: 'center',
    marginTop: 30,
    marginBottom: 10,
  },
  homeTitle: {
    color: '#fff',
    fontSize: 22,
    fontWeight: '300',
  }
})

export default PinText
