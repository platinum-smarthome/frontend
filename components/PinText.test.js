import React, { Component } from 'react'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import PinText from './PinText'
import { Text, View } from 'react-native'

enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<PinText />)
})

describe('<PinText/>', () => {
  it('should render <PinText/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<PinText/> child is rendered', () => {
  it('Should render <View/>', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('View').length).toBe(1)  
  })
  it('Should render <Text/>', () => {
    expect(wrapper.find(<Text/>)).toBeDefined()
    expect(wrapper.find('Text').length).toBe(1)  
  })
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<PinText/>)
    expect(tree).toMatchSnapshot()
  })
})