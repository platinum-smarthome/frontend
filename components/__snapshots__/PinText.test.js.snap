// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginBottom": 10,
      "marginTop": 30,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontSize": 22,
        "fontWeight": "300",
      }
    }
  />
</View>
`;
